presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/AbstractSqlInvokedFunctionNamespaceManager.java

```diff
@@ -174,6 +179,21 @@ public final ScalarFunctionImplementation getScalarFunctionImplementation(Functi
         return implementationByHandle.getUnchecked((SqlFunctionHandle) functionHandle);
     }
 
+    @Override
+    public CompletableFuture<Block> executeFunction(FunctionHandle functionHandle, Page input, List<Integer> channels, TypeManager typeManager)
+    {
+        checkArgument(functionHandle instanceof SqlFunctionHandle, format("Expect SqlFunctionHandle, got %s", functionHandle.getClass()));
+        FunctionMetadata functionMetadata = getFunctionMetadata(functionHandle);
+        ScalarFunctionImplementation functionImplementation = getScalarFunctionImplementation(functionHandle);
+        checkArgument(functionImplementation instanceof ThriftScalarFunctionImplementation, "Remote function execution currently only supports Thrift.");
+        return sqlFunctionExecutors.getThriftExecutor().executeFunction(
+                (ThriftScalarFunctionImplementation) functionImplementation,
```

Comment:
Can we expose a method `ScalarFunctionImplementation.getImplemenrationType()` and another method `SqlFunctionExecutors.getExecutor(ImplementationType)`? That way, we can avoid the check and avoid mentioning anything about thrift in this method.

Reply:
Why do we need to avoid mentioning thrift though?

Reply:
it's more about encapsulation. `SqlFunctionExecutors` currently is the class that knows about different types of executors, and their mapping from implementation type / language. Hence, it's better for other classes to not knowing about the executors - other classes only needs to know about implementation types, for example, and are able to execute functions via `SqlFunctionExecutors`.

Reply:
I added `executeFunction` to `SqlFunctionExecutors` instead.